html/data.php

```diff
@@ -7,24 +7,28 @@
  *
  * @copyright  (C) 2006 - 2012 Adam Armstrong
  */
+
+use App\Facades\DeviceCache;
+use App\Models\Port;
+
 $init_modules = ['web', 'auth'];
 require realpath(__DIR__ . '/..') . '/includes/init.php';
 
 if (is_numeric($_GET['id']) && (\LibreNMS\Config::get('allow_unauth_graphs') || port_permitted($_GET['id']))) {
-    $port = cleanPort(get_port_by_id($_GET['id']));
-    $device = device_by_id_cache($port['device_id']);
+    $port = cleanPort(Port::find($_GET['id']));
+    $device = DeviceCache::get($port->device_id);
     $title = generate_device_link($device);
     $title .= ' :: Port  ' . generate_port_link($port);
     $auth = true;
 
-    $in = snmp_get($device, 'ifHCInOctets.' . $port['ifIndex'], '-OUqnv', 'IF-MIB');
+    $in = snmp_get($device, 'ifHCInOctets.' . $port->ifIndex, '-OUqnv', 'IF-MIB');
```

Comment:
`$device` is an Object now but `snmp_get` requires an array. I haven't looked at the rest of the PR, can you please check for similar occurrences elsewhere?

Reply:
I think I already saw a Object related Class `SnmpQuery` seems to be more modern but I  need to dig in the sources first. I will convert it. A workaround would be `$device->toArray()` but it feels like a step back.